L1Trigger/TrackerTFP/src/LayerEncoding.cc

```diff
@@ -0,0 +1,94 @@
+#include "L1Trigger/TrackerTFP/interface/LayerEncoding.h"
+
+#include <vector>
+#include <set>
+#include <algorithm>
+#include <cmath>
+
+using namespace std;
+using namespace trackerDTC;
+
+namespace trackerTFP {
+
+  LayerEncoding::LayerEncoding(const DataFormats* dataFormats) :
+    dataFormats_(dataFormats),
+    setup_(dataFormats->setup()),
+    zT_(&dataFormats->format(Variable::zT, Process::sf)),
+    cot_(&dataFormats->format(Variable::cot, Process::sf)),
+    layerEncoding_(setup_->numSectorsEta(), vector<vector<vector<int>>>(pow(2, zT_->width()), vector<vector<int>>(pow(2, cot_->width())))),
+    maybeLayer_(setup_->numSectorsEta(), vector<vector<vector<int>>>(pow(2, zT_->width()), vector<vector<int>>(pow(2, cot_->width()))))
+  {
+    static constexpr int boundaries = 2;
+    vector<const SensorModule*> sensorModules;
+    sensorModules.reserve(setup_->sensorModules().size());
+    for (const SensorModule& sm : setup_->sensorModules())
+      sensorModules.push_back(&sm);
+    auto smallerR = [](const SensorModule* lhs, const SensorModule* rhs){ return lhs->r() < rhs->r(); };
+    auto smallerZ = [](const SensorModule* lhs, const SensorModule* rhs){ return lhs->z() < rhs->z(); };
+    auto equalRZ = [](const SensorModule* lhs, const SensorModule* rhs){ return abs(lhs->r() - rhs->r()) < 1.e-3 && abs(lhs->z() - rhs->z()) < 1.e-3; };
+    stable_sort(sensorModules.begin(), sensorModules.end(), smallerR);
+    stable_sort(sensorModules.begin(), sensorModules.end(), smallerZ);
+    sensorModules.erase(unique(sensorModules.begin(), sensorModules.end(), equalRZ), sensorModules.end());
+    for (int binEta = 0; binEta < setup_->numSectorsEta(); binEta++) {
+      const double sectorCot = (sinh(setup_->boundarieEta(binEta + 1)) + sinh(setup_->boundarieEta(binEta))) / 2.;
+      const double sectorZT = sectorCot * setup_->chosenRofZ();
+      const double rangeZT = (sinh(setup_->boundarieEta(binEta + 1)) - sinh(setup_->boundarieEta(binEta))) / 2. * setup_->chosenRofZ();
```

Comment:
In R.H.S. of L34-35, put "setup_->chosenRofZ() *" first, to make it clear, without worrying about operator ordering,  at L35 that chosenRofZ() is not in the denominator.

Reply:
done